### PR TITLE
Feature/bianca 14

### PR DESCRIPTION
**Pull request by bbiancab**
ewe

### CODE REVIEW

File: step-2/application/src/components/App/App.tsx [hunk 1]
```diff
@@ -76,20 +79,26 @@ export class App extends React.Component<IAppProps, AppState> {
...
-  public componentDidMount() {
-    this.props.fetchData();
-  }
-
   public render() {
     if (this.props.isLoading) {
-      return <Spin />;
+     return (
+     <div> 
+         <Spin />
```
Comment (LesleyMerks) [inline step-2/application/src/components/App/App.tsx]: double spinn?

File: step-2/application/src/components/App/App.tsx [hunk 2]
```diff
@@ -76,20 +79,26 @@ export class App extends React.Component<IAppProps, AppState> {
...
+         <Spin />
+       <Button  onClick={this.props.cancelFetch}  type="danger">Cancel</Button> 
+     
+     </div>);
+    //  return ;
```
Comment (LesleyMerks) [inline step-2/application/src/components/App/App.tsx]: ?

File: step-2/application/src/modules/Commits/Actions.ts [hunk 3]
```diff
@@ -33,6 +35,13 @@ const setSelectedId = (id: string) => {
   } as const;
 };
 
+
+export const fetchCancelled = () => {
+  return {
+    type: cancelo
```
Comment (LesleyMerks) [inline step-2/application/src/modules/Commits/Actions.ts]: typo

File: step-2/application/src/modules/Commits/Actions.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ export const loading = "FetchData_Loading";
 export const success = "FetchData_Success";
 export const failure = "FetchData_Failure";
 export const setId = "Set_Selected_Id";
+export const cancelo = "Cancelled";
```
Comment (LesleyMerks) [inline step-2/application/src/modules/Commits/Actions.ts]: typo

File: step-2/application/src/modules/Commits/reducer.ts [hunk 5]
```diff
@@ -55,6 +56,7 @@ export const commitReducer = (
         selectedIds: []
       };
     case success:
+      console.log("efwef");
```
Comment (LesleyMerks) [inline step-2/application/src/modules/Commits/reducer.ts]: can be removed

File: step-2/application/src/modules/Commits/sagas.ts [hunk 6]
```diff
@@ -1,28 +1,46 @@
...
 
+    
     yield put(fetchDataSuccess(response));
+    console.log("saga fetch success");
   } catch (error) {
     yield put(fetchDataFailure(error));
+  } finally{
+    if (yield cancelled()) {
+      console.log("saga canceled");
+      yield put(fetchCancelled());
```
Comment (LesleyMerks) [inline step-2/application/src/modules/Commits/sagas.ts]: if you open this in redux dev tools, it should call cancel twice, correct?

Comment (bbiancab) [reply step-2/application/src/modules/Commits/sagas.ts]: yes, it is

**Pull request merged**